test/cpp/end2end/ssl_credentials_test.cc

```diff
@@ -176,30 +403,165 @@ TEST_F(SslCredentialsTest, ConcurrentResumption) {
   grpc_ssl_session_cache_destroy(cache);
 }
 
-TEST_F(SslCredentialsTest, ResumptionFailsDueToNoCapacityInCache) {
+TEST_P(SslCredentialsTest, ResumptionFailsDueToNoCapacityInCache) {
   server_addr_ = absl::StrCat("localhost:",
                               std::to_string(grpc_pick_unused_port_or_die()));
   absl::Notification notification;
-  server_thread_ = new std::thread([&]() { RunServer(&notification); });
+  server_thread_ = new std::thread([&]() {
+    std::string root_cert = GetFileContents(kCaCertPath);
+    RunServer(&notification, root_cert);
+  });
   notification.WaitForNotification();
 
-  std::string root_cert = grpc_core::testing::GetFileContents(kCaCertPath);
-  std::string client_key = grpc_core::testing::GetFileContents(kClientKeyPath);
-  std::string client_cert =
-      grpc_core::testing::GetFileContents(kClientCertPath);
+  std::string root_cert = GetFileContents(kCaCertPath);
+  std::string client_key = GetFileContents(kClientKeyPath);
+  std::string client_cert = GetFileContents(kClientCertPath);
   grpc::SslCredentialsOptions ssl_options;
   ssl_options.pem_root_certs = root_cert;
   ssl_options.pem_private_key = client_key;
   ssl_options.pem_cert_chain = client_cert;
 
   grpc_ssl_session_cache* cache = grpc_ssl_session_cache_create_lru(0);
 
-  DoRpc(server_addr_, ssl_options, cache, /*expect_session_reuse=*/false);
-  DoRpc(server_addr_, ssl_options, cache, /*expect_session_reuse=*/false);
+  for (int i = 0; i < 2; ++i) {
+    auto full_handshake_context = DoRpc(ssl_options, cache);
+    EXPECT_EQ(full_handshake_context.status(), absl::OkStatus());
+    EXPECT_EQ(GetAuthContextProperty(**full_handshake_context,
+                                     GRPC_SSL_SESSION_REUSED_PROPERTY),
+              "false");
+  }
 
   grpc_ssl_session_cache_destroy(cache);
 }
 
+TEST_P(SslCredentialsTest, ServerCertificateIsUntrusted) {
+  server_addr_ = absl::StrCat("localhost:",
+                              std::to_string(grpc_pick_unused_port_or_die()));
+  absl::Notification notification;
+  server_thread_ = new std::thread([&]() {
+    std::string root_cert = GetFileContents(kCaCertPath);
+    RunServer(&notification, root_cert);
+  });
+  notification.WaitForNotification();
+
+  std::string root_cert = GetFileContents(kClientCertPath);
```

Comment:
This one too

Reply:
This is correct, because you don't want the client to trust the server's certificate. Added a comment to clarify.